app/models/cmr.rb

```diff
@@ -355,24 +355,45 @@ def self.get_raw_granule(concept_id)
     granule_results["Granule"]
   end
 
+  # Given a granule concept id, it will pull all revisions and return the latest granule revision.
   def self.get_raw_granule_results(concept_id)
+    url = Cmr.api_url("granules", "umm_json", {"concept_id" => concept_id})
+    granule_json_str = Cmr.cmr_request(url).parsed_response
+    granule_dict = JSON.parse(granule_json_str)
+    if granule_dict['hits'].to_i == 0
+      raise CmrError, "CMR returned 0 hits for #{concept_id}"
+    end
+    # Access to first element because it should have only one record (concept_id)
+    raw_format = granule_dict['items'][0]['meta']['format']
+    format_type = raw_format == 'application/vnd.nasa.cmr.umm+json' ? 'umm_json' : 'echo10'
+
+    if format_type == 'echo10'
+      granule_result_echo10 = get_raw_granule_results_echo10(concept_id)
+      granule_result_echo10['format_type'] = 'echo10'
+      return granule_result_echo10
+    else
+      granule_dict = granule_dict['items'][0]
+      granule_dict['format_type'] = format_type
+      granule_dict['concept-id'] = granule_dict['meta']['concept-id']
+      granule_dict['concept_id'] = granule_dict['meta']['concept-id']
+      granule_dict['revision_id'] = granule_dict['meta']['revision-id']
+      granule_dict['Granule'] = granule_dict['umm']
+    end
+    granule_dict['items'] = []
+    granule_dict
+  end
+
+  def self.get_raw_granule_results_echo10(concept_id)
```

Comment:
Likewise if this is going to return a single granule, shouldn't have _results in the name